universal-application-tool-0.0.1/app/auth/saml/LoginRadiusSamlProvider.java

```diff
@@ -0,0 +1,61 @@
+package auth.saml;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import auth.ProfileFactory;
+import com.google.inject.Inject;
+import javax.inject.Provider;
+import org.pac4j.core.http.callback.PathParameterCallbackUrlResolver;
+import org.pac4j.saml.client.SAML2Client;
+import org.pac4j.saml.config.SAML2Configuration;
+import repository.UserRepository;
+
+public class LoginRadiusSamlProvider implements Provider<SAML2Client> {
+
+  private final com.typesafe.config.Config configuration;
+  private final ProfileFactory profileFactory;
+  private final Provider<UserRepository> applicantRepositoryProvider;
+  private final String baseUrl;
+
+  @Inject
+  public LoginRadiusSamlProvider(
```

Comment:
praise: great work writing these provider classes! they look so clean and it's nice to have the logic specific to each IDP separated out into its own file.